### ISSUE TITLE
Exception when instantiating NoOpRavenClient

### ISSUE THREAD

**Issue by alirezajm** [getsentry/raven-csharp#230]
Instantiating a NoOpRaventClient will result in a Invalid Dsn exception.

here's the trace
```
EXCEPTION: System.ArgumentException: Invalid DSN
Parameter name: dsn ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at SharpRaven.Dsn.GetPrivateKey(Uri uri)
   at SharpRaven.Dsn..ctor(String dsn)
   --- End of inner exception stack trace ---
   at SharpRaven.Dsn..ctor(String dsn)
   at SharpRaven.NoOpRavenClient..ctor()

```

I tried to pass the `http://localhost/no-op-dsn` and `http://sentry-dsn.invalid` both are considered invalid. The issue is obviously from the Dsn class. 

_building for netcoreapp2.0_

**Comment by alirezajm** [comment]
Well it's trying to find credentials from the uri which aren't there. 
Can do a fix if wanted.

**Comment by alirezajm** [comment]
posted a simple fix: #231
